### DISCUSSION TITLE
Annotated Dependencies fail when used as route dependency

### DISCUSSION THREAD

**Discussion by ahmedabdou14** [fastapi/fastapi#9654] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Depends
from fastapi.security import OAuth2PasswordBearer
from typing import Annotated


oauth2_scheme = OAuth2PasswordBearer(tokenUrl=f"api/v1/auth/oauth2")
tokenDep = Annotated[str, Depends(oauth2_scheme)]

def get_current_admin(token: tokenDep):
   # logic
   return "admin"

getAdminDep = Annotated[str, Depends(get_current_admin)]

app = FastAPI()

@app.get("/", dependencies=[getAdminDep])
async def private_resaurce():
   return "private"
```

### Description

Using annotated dependencies inside of a route's dependency `app.get("/", dependencies=[getAdminDep])` fails.

### Operating System

Linux

### Operating System Details

Ubuntu 22.04

### FastAPI Version

0.96.1

### Python Version

3.11

**Comment by wu-clan** [comment]
Hi, Annotated should be used as a parameter comment, in routing you should use Depends directly, try:

```python
getAdminDep = Depends(get_current_admin)

app.get("/", dependencies=[getAdminDep])
async def private_resaurce():
   return "private"
```

**Reply by ahmedabdou14** [reply]
Yes, ofcourse this works, but shouldn't it support annotated dependencies aswell? This would provide consistency throughout the project.

**Reply by wu-clan** [reply]
fastapi resolves Annotated parameters internally, but for dependencies it does not have additional resolution, so maybe you can create issues and ask for new features

**Reply by wu-clan** [reply]
You can check here：#4871

**Reply by ahmedabdou14** [reply]
Sounds good. I am not really familiar with the protocols here, shall i just comment under that merged PR https://github.com/tiangolo/fastapi/pull/4871? or shall I create a new issue?

**Reply by wu-clan** [reply]
I think this will be a new problem, because 4871 is not essentially the same thing as dependencies, which are defined internally, while 4871 is specific to user input

**Reply by ahmedabdou14** [reply]
Apparently only Sebastian or delegates can create issues :) so I will tag Sebastian and let's see where can we take this.
Thank you for your follow up @wu-clan.

**Comment by ahmedabdou14** [comment]
@tiangolo please take a look at this discussion so that we can move it into issues if it is considered one (if not a feature request).

**Comment by Kludex** [comment, marked as answer]
Same as https://github.com/tiangolo/fastapi/discussions/9674#discussioncomment-6182123.
